runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/provider/MuleExtensionModelDeclarer.java

```diff
@@ -849,13 +873,16 @@ private void declareParallelForEach(ExtensionDeclarer extensionDeclarer, ClassTy
         .withDisplayModel(DisplayModel.builder().displayName(TARGET_VALUE_PARAMETER_DISPLAY_NAME).build())
         .withLayout(LayoutModel.builder().tabName(ADVANCED_TAB).build())
         .withModelProperty(new TargetModelProperty());
+
+    parallelForeach.withOutput().ofDynamicType(BaseTypeBuilder.create(MetadataFormat.JAVA).arrayType().of(ANY_TYPE).build());
+    parallelForeach.withOutputAttributes().ofDynamicType(ANY_TYPE);
```

Comment:
is it any or is it void? does it retain whatever attributes entered the parallel-foreach or does it nullify it?

Reply:
As far as I know, it keeps the attributes and combines them in case there is more than one